pkg/channel/get.go

```diff
@@ -17,18 +17,117 @@ limitations under the License.
+	"context"
+	"fmt"
+
 	"github.com/spf13/cobra"
+	corev1 "k8s.io/api/core/v1"
+	v1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"k8s.io/kubectl/pkg/cmd/get"
+
+	"github.com/bestchains/bc-cli/pkg/common"
 )
 
+// NewChanGetCmd returns the `kubectl get` command for channel.
 func NewChanGetCmd(option common.Options) *cobra.Command {
-	//defaultPrintFlag := get.NewGetPrintFlags()
+
+	defaultPrintFlag := get.NewGetPrintFlags()
+
 	cmd := &cobra.Command{
-		Use: "",
+		Use: "channel [NAME] -n NETWORK-NAME",
 		RunE: func(cmd *cobra.Command, args []string) error {
+			// get config from getter
+			cli, err := common.GetDynamicClient()
+			if err != nil {
+				fmt.Fprintln(option.ErrOut, err)
+				return err
+			}
+			list := corev1.List{
+				TypeMeta: v1.TypeMeta{
+					Kind:       "List",
+					APIVersion: "v1",
+				},
+				ListMeta: v1.ListMeta{},
+			}
+			var obj runtime.Object
+
+			netName, _ := cmd.Flags().GetString("network")
+
+			if netName == "" {
+				fmt.Fprintln(option.ErrOut, err)
+				return err
+			}
+
+			if len(args) == 0 {
+				// list all channel of the given network
+				net, err := cli.Resource(schema.GroupVersionResource{
+					Group:    common.IBPGroup,
+					Version:  common.IBPVersion,
+					Resource: common.NetworkResource,
+				}).Get(context.TODO(), netName, v1.GetOptions{})
+				if err != nil {
+					fmt.Fprintln(option.ErrOut, err)
+					return err
+				}
+
+				channels, _, _ := unstructured.NestedStringSlice(net.Object, "status", "channels")
+
+				for _, chName := range channels {
+					ch, err := cli.Resource(schema.GroupVersionResource{
+						Group:    common.IBPGroup,
+						Version:  common.IBPVersion,
+						Resource: common.Channel,
+					}).Get(context.TODO(), chName, v1.GetOptions{})
+					if err != nil {
+						fmt.Fprintln(option.ErrOut, err)
+						return err
+					}
+					list.Items = append(list.Items, runtime.RawExtension{Object: ch})
+				}
+
+			} else {
+				// get channel by name
+				for _, arg := range args {
+					ch, err := cli.Resource(schema.GroupVersionResource{
+						Group:    common.IBPGroup,
+						Version:  common.IBPVersion,
+						Resource: common.Channel,
+					}).Get(context.TODO(), arg, v1.GetOptions{})
+					if err != nil {
+						fmt.Fprintln(option.ErrOut, err)
+						continue
+					}
+					list.Items = append(list.Items, runtime.RawExtension{Object: ch})
+				}
+			}
+
+			// object process
+			if len(list.Items) != 1 {
+				obj, err = common.ListToObj(list)
+				if err != nil {
+					fmt.Fprintln(option.ErrOut, err)
+					return err
+				}
+			} else {
+				obj = list.Items[0].Object
+			}
+
+			// print result
+			p, err := defaultPrintFlag.ToPrinter()
+			if err != nil {
+				fmt.Fprintln(option.ErrOut, err)
+				return err
+			}
+			_ = p.PrintObj(obj, option.Out)
 			return nil
 		},
 	}
 
+	_ = cmd.MarkFlagRequired("network")
```

Comment:
```go
cmd.Flags().StringP("network", "n", "", "network of the desired channel")
_ = cmd.MarkFlagRequired("network")
```
After adding the flags, then set it as a  required parameter.  
When set as a required parameter, network must not be empty.  we can delete the if statement in line 59.

Reply:
@Lanture1064 have you tested this CLI commands before you make this PR ?